web/src/pages/AdminPage/UserEditor.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    TextField,
+} from '@mui/material'
+import * as React from 'react'
+import { useEffect, useState } from 'react'
+import { User, UserUpdateRequest } from '../../model/user'
+
+interface UserEditorProps {
+    user: User | null
+    onSave: (userId: number, newUser: UserUpdateRequest) => Promise<void>
+    onCancel: () => void
+}
+
+const UserEditor: React.FC<UserEditorProps> = (props: UserEditorProps) => {
```

Comment:
Type annotation on `props: UserEditorProps` not needed, already specified in `React.FC`